scripts/dumps/dump_genome_from_core.py

```diff
@@ -0,0 +1,184 @@
+
+    Args:
+        host: The DB host name
+        port: The port number
+        core_db: The core_db name
+        fasta_filename: The name given for the FASTA file
+        mask: The mask format for the FASTA file.
+
+    Returns:
+        A subprocess call object to download the FASTA file using the PERL script.
+
+    """
+
+    work_dir = "{}/../master/ensembl-compara/scripts/dumps".format(
+        os.environ["ENSEMBL_ROOT_DIR"]
+    )
+    script = "dump_genome_from_core.pl"
+
+    perl_call = [
+        "perl",
+        "{}/{}".format(work_dir, script),
+        "--core_db",
+        "{}".format(core_db),
+        "--host",
+        "{}".format(host),
+        "--port",
+        "{}".format(port),
+        "--mask",
+        "{}".format(mask),
+        "--outfile",
+        "{}".format(fasta_filename),
+    ]
+    return subprocess_call(command=perl_call, ibsub=True, shell=False)
+
+
+def query_coredb(host, core_db, query):
+    """Get the correct meta production name
+
+    Args:
+        host: The DB host name
+        core_db: The core_db name
+
+    Returns:
+        A subprocess call object to query the database.
+
+    """
+    mysql_call = [
+        "{}".format(host),
+        "{}".format(core_db),
+        "-ss",
+        "-e",
+        "{}".format(query),
+    ]
+    return subprocess_call(command=mysql_call)
+
+
+def parse_yaml(file, dest):
+    """YAML parser.
+
+    Args:
+        file: The file object
+
+        dest: The destination PATH where the FASTA file will be stored
+    """
+    content = yaml.load(file, Loader=SafeLoader)
+    for data in content:
+
+        host = data["host"]
+        port = data["port"]
+
+        for core_db in data["core_db"]:
+            specie_name = query_coredb(
+                host=host,
+                core_db=core_db,
+                query='SELECT meta_value FROM meta WHERE meta_key="species.production_name";',
+            )
+
+            # in case gca is presented in the specie name
+            if "gca" not in specie_name:
+                gca_number = query_coredb(
+                    host=host,
+                    core_db=core_db,
+                    query='SELECT meta_value FROM meta WHERE meta_key="assembly.accession";',
+                )
+                # fix the name
+                gca_number = gca_number.replace(".", "v").replace('_','').lower()
+                specie_name = "{}_{}".format(specie_name, gca_number)
+
+            if specie_name is not None:
+                download_file(
+                    host=host,
+                    port=port,
+                    core_db=core_db,
+                    fasta_filename="{}/{}.fa".format(dest, specie_name),
+                )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--yaml", required=True, type=str, help="YAML input file")
+    parser.add_argument(
+        "--output", required=False, default=None, type=str, help="Processed output file"
+    )
```

Comment:
Unless there is an overriding reason, I would omit arguments if they are being assigned their default values.
I've also suggested changing the help string for the `--output` option to be explicit that it is a directory.
```suggestion
    parser.add_argument("--yaml", required=True, help="YAML input file")
    parser.add_argument("--output", help="Output directory.")
```